ProjFS.Mac/PrjFSLib/prjfs-log/kext-perf-tracing.cpp

```diff
@@ -90,16 +181,33 @@ bool PrjFSLog_FetchAndPrintKextProfilingData(io_connect_t connection)
                 uint64_t meanNS = numSamples > 0 ? sumNS / numSamples : 0;
 
                 printf(
-                    "[%15llu][%10llu][%10llu][%10llu]\n",
+                    "[%15llu][%10llu][%10llu][%10llu]",
                     sumNS,
                     meanNS,
                     nanosecondsFromAbsoluteTime(counters[i].min),
                     nanosecondsFromAbsoluteTime(counters[i].max));
+                
+                static const char* const barGraphItems[9] = {
+                    NBSP_STR, "▁", "▂", "▃", "▄", "▅", "▆", "▇", "█",
+                };
+                
+                _Atomic uint64_t(&buckets)[64] = counters[i].sampleBuckets;
+                uint64_t bucketMax = *std::max_element(begin(buckets), end(buckets));
+                if (bucketMax > 0) // Should normally not be 0 if we get here, but defends against divide by 0 in case of a bug
+                {
+                    printf("[");
+                    for (unsigned bucket = 0; bucket < 64; ++bucket)
+                    {
+                        // Always round up so we have a clear distinction between buckets with zero and even a single item.
+                        uint64_t eighths = (8u * buckets[bucket] + bucketMax - 1) / bucketMax;
```

Comment:
In this particular instance, the suffix isn't needed, as the signed int literals will be promoted to `uint64_t` anyway, so I'll just remove it.

I specifically used `UINT64_C` in the shift expression because the result of the expression absolutely needed 64 bits. `ull` would I guess be OK as we're not attempting to be cross-platform. I guess I tend to use the `stdint.h` macros when I want a specific type, and I'll use `u` if I want to force an unsigned type, but I don't particularly care about the rank.